model/src/main/kotlin/utils/Extensions.kt

```diff
@@ -143,6 +143,7 @@ enum class PurlType(private val value: String) {
     DEBIAN("deb"),
     DRUPAL("drupal"),
     GEM("gem"),
+    GENERIC("generic"),
```

Comment:
Commit message: Please link directly to https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#generic

Reply:
Also note that we've historically been using `a-name` (see further up in the `enum`) for more or less the same purpose. However, `a-name` is a [NexusIq / Sonatype specific thing](https://help.sonatype.com/iqserver/product-information/referencing-package-url-%28purl%29-and-component-identifiers#ReferencingPackageURL(purl)andComponentIdentifiers-a-name). I'll probably do a follow-up PR with docs to explain this.